src/build.ts

```diff
@@ -229,8 +229,12 @@ async function _build(
       entry.builder = entry.input.endsWith("/") ? "mkdist" : "rollup";
     }
 
-    if (options.declaration !== undefined && entry.declaration === undefined) {
-      entry.declaration = options.declaration;
+    if (
+      (entry.builder === "mkdist" || entry.builder === "untyped") &&
+      options.declaration !== undefined &&
+      entry.declaration === undefined
+    ) {
+      entry.declaration = !!options.declaration;
```

Comment:
Let's keep runtime simple. it is simply an additional feature of mkdist

Reply:
> keep runtime simple.

Otherwise, we’d need `@ts-expect-error` to suppress type errors...

> used by mkdist

Not just mkdist, untyped also uses it: https://github.com/unjs/unbuild/blob/main/src/builders/untyped/index.ts#L70